app/models/header_lookup.rb

```diff
@@ -0,0 +1,106 @@
+class HeaderLookup
+  BASE_SPARQL_MESH_URI ="https://id.nlm.nih.gov/mesh/sparql?format=JSON&limit=10&inference=true&query=PREFIX%20rdfs"\
+    "%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0D%0APREFIX%20meshv%3A%20%3Chttp%3A%2F%2Fid.nlm."\
+    "nih.gov%2Fmesh%2Fvocab%23%3E%0D%0APREFIX%20mesh2018%3A%20%3Chttp%3A%2F%2Fid.nlm.nih.gov%2Fmesh%3E%0D%0A%0D%0ASELECT"\
+    "%20%3Fd%20%3FdName%0D%0AFROM%20%3Chttp%3A%2F%2Fid.nlm.nih.gov%2Fmesh%3E%0D%0AWHERE%20%7B%0D%0A%20%20%3Fd%20a%20"\
+    "meshv%3ADescriptor%20.%0D%0A%20%20%3Fd%20rdfs%3Alabel%20%3FdName%0D%0A%20%20FILTER(REGEX(%3FdName%2C%27"
+  END_SPARQL_MESH_URI = "%27%2C%20%27i%27))%20%0D%0A%7D%20%0D%0AORDER%20BY%20%3Fd%20%0D%0A"
+  LCSH_BASE_URI = "http://id.loc.gov/authorities/subjects/suggest/?q="
+  MEMOIZED_MESH_FILE = "memoized_mesh.txt"
+  MEMOIZED_LCSH_FILE = "memoized_lcsh.txt"
+
+  def initialize(memoized_mesh_file_path=MEMOIZED_MESH_FILE, memoized_lcsh_file_path=MEMOIZED_LCSH_FILE)
+    @@memoized_mesh ||= read_memoized_headers(memoized_mesh_file_path)
```

Comment:
`||=` is a really helpful operator in ruby. It only assigns the value if it has not been assigned already.

Reply:
I agree, I these are really great little operators.